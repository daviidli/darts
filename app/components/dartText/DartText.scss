.double::after {
	content: 'x2';
	font-size: 0.85em;
	padding-left: 0.25em;
	color: #e74c3c;
}

.triple::after {
	content: 'x3';
	font-size: 0.85em;
	padding-left: 0.25em;
	color: #3498db;
}
