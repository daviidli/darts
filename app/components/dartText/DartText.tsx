import React from 'react';
import clsx from 'clsx';
import { dartValue } from '../../utils/dartUtils';
import styles from './DartText.scss';

type Props = {
	value: dartValue;
	style?: {};
};

const DartText = (props: Props) => {
	const { value, style } = props;
	return (
		<span
			className={clsx({
				[styles.double]: value.modifier === 'd',
				[styles.triple]: value.modifier === 't'
			})}
			style={style}
		>
			{value.value}
		</span>
	);
};

export default DartText;
