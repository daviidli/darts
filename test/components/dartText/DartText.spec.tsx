import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';
import DartText, { Props } from '../../../app/components/dartText/DartText';

Enzyme.configure({ adapter: new Adapter() });

describe('Options component', () => {
	it('should match snapshot', () => {
		const props: Props = {
			value: {
				value: '20',
				modifier: 't'
			}
		};
		const tree = renderer.create(<DartText {...props} />).toJSON();
		expect(tree).toMatchSnapshot();
	});

	it('should match snapshot without a modifier', () => {
		const props: Props = {
			value: {
				value: '20',
				modifier: ''
			}
		};
		const tree = renderer.create(<DartText {...props} />).toJSON();
		expect(tree).toMatchSnapshot();
	});
});
