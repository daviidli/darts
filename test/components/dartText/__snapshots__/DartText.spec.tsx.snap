// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Options component should match snapshot 1`] = `
<span
  className="triple"
>
  20
</span>
`;

exports[`Options component should match snapshot without a modifier 1`] = `
<span
  className=""
>
  20
</span>
`;
